src/features/commits/DatasetCommitList.tsx

```diff
@@ -45,6 +47,19 @@ const DatasetCommits: React.FC<DatasetCommitsProps> = ({
           <HistorySearchBox />
           {editable && <NewVersionButton qriRef={qriRef} />}
         */}
+        <li className='flex items-stretch text-black tracking-wider'>
+          <div className='relative w-4 mr-5 flex-shrink-0'>
+            <div className={classNames('absolute top-2.5 w-4 h-4 rounded-3xl border border-qritile-600')}>&nbsp;</div>
+            <div className='relative line-container w-0.5 mx-auto h-full'>
+              <div className='absolute -bottom-2 w-full h-11 bg-gray-300 rounded'>&nbsp;</div>
+            </div>
+          </div>
+          <div className='w-full'>
+            <Link to={`/${qriRef.username}/${qriRef.name}/edit`}>
+              <Button className='mb-6' block disabled={loading || (commits.length ? path !== commits[0].path : false)} type='primary-outline' icon='edit'>Edit</Button>
```

Comment:
we should have some check here if the user is logged in and it is their dataset, only then do we should the `Edit` button